testutil/fuse/fuse.go

```diff
@@ -0,0 +1,130 @@
+// +build linux darwin
+
+// Copyright 2019 The qiffang Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fuse
+
+import (
+	"fmt"
+	"github.com/prometheus/tsdb/testutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/hanwen/go-fuse/fuse"
+	"github.com/hanwen/go-fuse/fuse/nodefs"
+	"github.com/hanwen/go-fuse/fuse/pathfs"
+)
+
+type loopBackFs struct {
+	pathfs.FileSystem
+}
+
+// A FUSE filesystem that shunts all request to an underlying file system.
+// Its main purpose is to provide test coverage.
+type hookFs struct {
```

Comment:
look here for documenting exported and unexperted objects
blog.golang.org/godoc-documenting-go-code

the comment should always start with the name of the object
`// hookFs .....`

this is because of the auto generate docs.

Reply:
Accept and I will updated it.
Thanks